### PR TITLE
Change SmtpCommandType to interface

### PR DESCRIPTION
**Pull request by cifer828**
## Description

- Change SmtpCommandType to interface
- Add SmtpRFCSupportedCommandType enum to represent existing SmtpCommandType enum

## Motivation and Context

Jedi needs support custom non-RFC command to talk to Proxy team, so the SmtpCommandType should include both RFC and non-RFC commands.

## How Has This Been Tested?

Use previous unit test of ExtendedHelloCommand, HelloCommand and QuitCommand

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/main/java/com/yahoo/smtpnio/async/request/SmtpRFCSupportedCommandType.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.yahoo.smtpnio.async.request;
```
Comment (fansu) [inline core/src/main/java/com/yahoo/smtpnio/async/request/SmtpRFCSupportedCommandType.java:1]: Please follow other source code and add those license info at the beginning.

File: core/src/test/java/com/yahoo/smtpnio/async/request/ExtendedHelloCommandTest.java [hunk 2]
```diff
@@ -66,9 +66,10 @@ public void testGetCommandLine() throws SmtpAsyncClientException, IllegalAccessE
      */
     @Test
     public void testGetCommandType() {
-        Assert.assertSame(new ExtendedHelloCommand("test.client.me").getCommandType(), SmtpCommandType.EHLO,
+        Assert.assertSame(new ExtendedHelloCommand("test.client.me").getCommandType(),
+                SmtpRFCSupportedCommandType.EHLO,
```
Comment (fansu) [inline core/src/test/java/com/yahoo/smtpnio/async/request/ExtendedHelloCommandTest.java:70]: minor: fix line 71 to be same as line 70.

**Comment by fansu** [comment]
Check the comments above

File: core/src/main/java/com/yahoo/smtpnio/async/request/SmtpRFCSupportedCommandType.java (hunk 1 above)
Comment (cifer828) [reply core/src/main/java/com/yahoo/smtpnio/async/request/SmtpRFCSupportedCommandType.java:1]: Will add license info.

File: core/src/test/java/com/yahoo/smtpnio/async/request/ExtendedHelloCommandTest.java (hunk 2 above)
Comment (cifer828) [reply core/src/test/java/com/yahoo/smtpnio/async/request/ExtendedHelloCommandTest.java:70]: is it ok to just combine 69,70,71 into one line? The style checker will automatically combine them when saving the file.

Comment (fansu) [reply core/src/test/java/com/yahoo/smtpnio/async/request/ExtendedHelloCommandTest.java:70]: yes. thats fine. as long that line not past max character in single line for checkstyle

**Comment by fansu** [comment]
@nituyahoo can you take a look at this change?

**Pull request merged**